lib/rubocop/feature_loader.rb

```diff
@@ -30,14 +39,12 @@ def initialize(config_directory_path:, feature:)
     end
 
     def load
-      # Don't use `::Kernel.require(target)` to prevent the following error:
-      # https://github.com/rubocop/rubocop/issues/10893
-      require(target)
+      self.class.call_require(target)
     rescue ::LoadError => e
       raise if e.path != target
 
       begin
-        ::Kernel.require(namespaced_target)
+        self.class.call_require(namespaced_target)
```

Comment:
Implementations don't want to be complicated for stub/mock testing. Can you update it to use `require` and testing?

```suggestion
        require(namespaced_target)
```

Reply:
I agree with you on that point, but we are using `RSpec/AnyInstance`. If we use `require` directly, how should we test without it?

Reply:
Just to be clear, this singleton method was not necessarily prepared only for stub. This pull request was made because I used `Kernel.require` in two places and forgot to change one of them. If so, I thought it would be good to keep one caller as a preventive measure.

Reply:
`rubocop:disable` comment could be used. There may be a better way, but I won't know soon.

```diff
diff --git a/spec/rubocop/feature_loader_spec.rb b/spec/rubocop/feature_loader_spec.rb
index 4ac7cd102..0a7a00d68 100644
--- a/spec/rubocop/feature_loader_spec.rb
+++ b/spec/rubocop/feature_loader_spec.rb
@@ -1,6 +1,10 @@
 # frozen_string_literal: true

 RSpec.describe RuboCop::FeatureLoader do
+  # rubocop:disable RSpec/AnyInstance
+  let(:feature_loader) { allow_any_instance_of(described_class) }
+  # rubocop:enable RSpec/AnyInstance
+
   describe '.load' do
     subject(:load) do
       described_class.load(config_directory_path: config_directory_path, feature: feature)
@@ -16,18 +20,18 @@ RSpec.describe RuboCop::FeatureLoader do

     context 'with normally lodable feature' do
       before do
-        allow(Kernel).to receive(:require)
+        feature_loader.to receive(:require)
       end
```

Reply:
Does that mean we should change it like this?

```diff
diff --git a/lib/rubocop/feature_loader.rb b/lib/rubocop/feature_loader.rb
index 39e40b57c..af22a2ab8 100644
--- a/lib/rubocop/feature_loader.rb
+++ b/lib/rubocop/feature_loader.rb
@@ -20,15 +20,6 @@ module RuboCop
       def load(config_directory_path:, feature:)
         new(config_directory_path: config_directory_path, feature: feature).load
       end
-
-      # @note Don't use `::Kernel.require(target)` here.
-      #   To prevent the following error:
-      #   https://github.com/rubocop/rubocop/issues/10893
-      # @api private
-      # @param [String] target
-      def call_require(target)
-        require(target)
-      end
     end
 
     # @param [String] config_directory_path
@@ -39,12 +30,16 @@ module RuboCop
     end
 
     def load
-      self.class.call_require(target)
+      # Don't use `::Kernel.require(target)` to prevent the following error:
+      # https://github.com/rubocop/rubocop/issues/10893
+      require(target)
     rescue ::LoadError => e
       raise if e.path != target
 
       begin
-        self.class.call_require(namespaced_target)
+        # Don't use `::Kernel.require(target)` to prevent the following error:
+        # https://github.com/rubocop/rubocop/issues/10893
+        require(namespaced_target)
       rescue ::LoadError => error_for_namespaced_target
         raise e if error_for_namespaced_target.path == namespaced_target
 
diff --git a/spec/rubocop/feature_loader_spec.rb b/spec/rubocop/feature_loader_spec.rb
index 1436aee62..c247bd37a 100644
--- a/spec/rubocop/feature_loader_spec.rb
+++ b/spec/rubocop/feature_loader_spec.rb
@@ -14,20 +14,28 @@ RSpec.describe RuboCop::FeatureLoader do
       'feature'
     end
 
+    let(:allow_feature_loader) do
+      allow_any_instance_of(described_class) # rubocop:disable RSpec/AnyInstance
+    end
+
+    let(:expect_feature_loader) do
+      expect_any_instance_of(described_class) # rubocop:disable RSpec/AnyInstance
+    end
+
     context 'with normally lodable feature' do
       before do
-        allow(described_class).to receive(:call_require)
+        allow_feature_loader.to receive(:require)
       end
 
       it 'loads it normally' do
-        expect(described_class).to receive(:call_require).with('feature')
+        expect_feature_loader.to receive(:require).with('feature')
         load
       end
     end
 
     context 'with dot-prefixed lodable feature' do
       before do
-        allow(described_class).to receive(:call_require)
+        allow_feature_loader.to receive(:require)
       end
 
       let(:feature) do
@@ -35,15 +43,15 @@ RSpec.describe RuboCop::FeatureLoader do
       end
 
       it 'loads it as relative path' do
-        expect(described_class).to receive(:call_require).with('path-to-config/./path/to/feature')
+        expect_feature_loader.to receive(:require).with('path-to-config/./path/to/feature')
         load
       end
     end
 
     context 'with namespaced feature' do
       before do
-        allow(described_class).to receive(:call_require).with('feature-foo').and_call_original
-        allow(described_class).to receive(:call_require).with('feature/foo')
+        allow_feature_loader.to receive(:require).with('feature-foo').and_call_original
+        allow_feature_loader.to receive(:require).with('feature/foo')
       end
 
       let(:feature) do
@@ -51,16 +59,16 @@ RSpec.describe RuboCop::FeatureLoader do
       end
 
       it 'loads it as namespaced feature' do
-        expect(described_class).to receive(:call_require).with('feature/foo')
+        expect_feature_loader.to receive(:require).with('feature/foo')
         load
       end
     end
 
     context 'with dot-prefixed namespaced feature' do
       before do
-        allow(described_class).to receive(:call_require).with('path-to-config/./feature-foo')
-                                                        .and_call_original
-        allow(described_class).to receive(:call_require).with('path-to-config/./feature/foo')
+        allow_feature_loader.to receive(:require).with('path-to-config/./feature-foo')
+                                                 .and_call_original
+        allow_feature_loader.to receive(:require).with('path-to-config/./feature/foo')
       end
 
       let(:feature) do
@@ -68,14 +76,14 @@ RSpec.describe RuboCop::FeatureLoader do
       end
 
       it 'loads it as namespaced feature' do
-        expect(described_class).to receive(:call_require).with('path-to-config/./feature/foo')
+        expect_feature_loader.to receive(:require).with('path-to-config/./feature/foo')
         load
       end
     end
 
     context 'with unexpected LoadError from require' do
       before do
-        allow(described_class).to receive(:call_require).and_raise(LoadError)
+        allow_feature_loader.to receive(:require).and_raise(LoadError)
       end
 
       it 'raises LoadError' do
```

Reply:
Can you apply it to this PR?